### PR TITLE
Implementing pa cccisa

### PR DESCRIPTION
**Pull request by gv-ORC**
I now have the base instruction set for Processor Academy Crash Course CPU defined using my framework.. the assemble and disassemble functions, as well as the execution functions for LoadImmediate, Halt, Jump, and Branch still need to be completed... The later of the 3 will be held off until I get the emulator side of things properly interfacing with the instruction set format.

### CODE REVIEW

**Comment by gv-ORC** [comment]
I feel like I should have 100% squashed these commits.. would have shown all the edits to the single file in 1 quick review.

Is that correct?

File: public/assembler_playground.js [hunk 1]
```diff
@@ -108,66 +100,241 @@ class EmulatedInstruction {
...
+    constructor (mnemonic, instructionAssembly, instructionOperations) {
+        this.mnemonic = mnemonic;
+        this.instructionAssembly = instructionAssembly;
+        this.instructionOperations = instructionOperations;
+    }
+}
+
+//? Format Definitions
+// Operands
+const operandA = new InstructionOperand("A", "Dest-Register", undefined, 11, 8);
```
Comment (Myvar) [inline public/assembler_playground.js:112]: Make things like this an enum:
"Dest-Register"

when ever a string is known and not dynamic we use an enum(you can easily load an enum from json)

File: public/assembler_playground.js [hunk 2]
```diff
@@ -108,66 +100,241 @@ class EmulatedInstruction {
...
+
+}
+function disassembleNop(line) {
+    const assembledLine = line;
+    return assembledLine;
+    
+}
+// Three Operand
+const threeOperandMachineFormat = new InstructionFormat("three-operand", 16, majorOpcode, [], [operandB, operandC], [operandA]);
+function assembleThreeOperand(line) {
```
Comment (Myvar) [inline public/assembler_playground.js:137]: Why are these functions out here?

File: public/assembler_playground.js [hunk 3]
```diff
@@ -108,66 +100,241 @@ class EmulatedInstruction {
...
+function subExecution (instruction, emulatorState) {
+    return  sourceOperands[0] - sourceOperands[1];
 }
-const subAssembly = new Generic_InstructionAssembly(16, [subOpcode, OperandA, OperandB, OperandC], []);
 const subOperations = new Generic_Operation("stage_0", subExecution);
 
-// Brining together the ISA
+// AND
+const andAssembly = new InstructionAssembly(threeOperandMachineFormat, assembleThreeOperand, disassembleThreeOperand);
+function andExecution (instruction, emulatorState) {
```
Comment (Myvar) [inline public/assembler_playground.js:235]: Should these not be overriding in a class?

**Pull request merged**